src/main/java/org/swisspush/redisques/handler/RedisquesHttpRequestHandler.java

```diff
@@ -291,18 +292,16 @@ private void getSingleLock(RoutingContext ctx) {
                 }));
     }
 
-    private Optional<String> decodedQueueNameOrRespondWithBadRequest(RoutingContext ctx, String encodedQueueName) {
-        try {
-            return Optional.of(java.net.URLDecoder.decode(encodedQueueName, StandardCharsets.UTF_8.name()));
-        } catch (UnsupportedEncodingException ex) {
-            // not going to happen - value came from JDK's own StandardCharsets
-            respondWith(StatusCode.BAD_REQUEST, ex.getMessage(), ctx.request());
-            return Optional.empty();
+    private Optional<String> decodedQueueNameOrRespondWithBadRequest(String encodedQueueName) {
+        if (enableQueueNameDecoding) {
+            return Optional.of(java.net.URLDecoder.decode(encodedQueueName, StandardCharsets.UTF_8));
+        } else {
+            return Optional.of(encodedQueueName);
```

Comment:
Why do we drop exception handling?

Reply:
Before the PR it was `UTF_8.name()` a string, which could have led to an UnsupportedEncodingException during interpretation. Now we use `UTF_8` which can’t. 
String, see: https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/URLDecoder.html#decode(java.lang.String,java.lang.String)

Charset, see: https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/URLDecoder.html#decode(java.lang.String,java.nio.charset.Charset)

Reply:
Hehe :) I inspected source code from the wrong jdk (jdk8) where this [overload seems not to exist](https://docs.oracle.com/javase/8/docs/api/java/net/URLDecoder.html#method.summary).

(*SideNote: So looks as this change, will make redisques unable to run with jdk8 (aka not backward-compatible)*)

Reply:
This redisques is already running on Java 11, and I don't think that we ever want to go back to Java 8.

I will have a look at these other exceptions, though, because AFAICT we don't seem to cover them (even before my changes)